ambry-api/src/main/java/com.github.ambry/store/Store.java

```diff
@@ -54,6 +54,13 @@
    */
   void delete(MessageWriteSet messageSetToDelete) throws StoreException;
 
+  /**
+   * Undelete the blob identified by {@code id}.
+   * @param info The {@link MessageInfo} that carries some basic information about this operation.
+   * @return the lifeVersion of the undeleted blob.
+   */
+  short undelete(MessageInfo info) throws StoreException;
```

Comment:
Any reason why we don't pass in `MessageWriteSet`? Is this because we undelete only one blob at a time?

Reply:
MessgeWriteSet includes a serialized undelete record, which will be persisted on disk in the log. The problem is that this serialized record requires a lifeVersion, but we don't know the lifeVersion before we query the index. So here I pass a MessageInfo instead of MessageWriteSet. You can see in the BlobStore.undelete method, I create a MessageWriteSet in the end, with a lifeVersion.